src/Sparrow.Server/Utils/UnguardedDisposableScope.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Sparrow.Server.Utils;
+
+public sealed class UnguardedDisposableScope : IDisposable
```

Comment:
The usage of `Unguarded` will make that in case of any errors we won't dispose all registered disposables. That is potentially the source of leaks

Reply:
It is, but that's why it has to be used properly (hot paths) and that losing the dispose is not a big deal.